src/api/api/BIOMD/BIOMD.js

```diff
@@ -416,3 +423,171 @@ module.exports.LIST_VENDORS = async function (req, dbClient) {
+    );
+
+    //Copy Requester Information
+    var res = Object.assign({}, req);
+
+    //req object defintion
+    var find = Object.assign({}, req.Request.find);
+
+    var reportType = find.report;
+
+    // Check report type (This should be included in the request inside find)
+    if (REPORT_TYPES.includes(reportType)) {
+      const ACTIVE_STRING = "Active Deployed";
+      const collection = await dbClient.db(instituteCode).collection("Asset");
+      var result;
+
+      switch(reportType) {
+        case DESCRIPTION: {
+          result = await collection.aggregate([
+            {
+              $project: {
+                assetCode: 1, 
+                description: 1,
+                purchaseCost: 1,
+                activeCount: {
+                  $cond: [{$eq: ["$status", ACTIVE_STRING]}, 1, 0]
+                },
+                inactiveCount: {
+                  $cond: [{$ne: ["$status", ACTIVE_STRING]}, 1, 0]
+                }
+              }
+            },
+            {
+              $group: {
+                _id: "$description",
+                assetCode: {"$first":"$assetCode"},
+                active: {$sum: "$activeCount"},
+                inactive: {$sum: "$inactiveCount"},
+                totalCost: {$sum: "$purchaseCost"}
+              }
+            }
+          ]).toArray();
+          break;
+        }
+        case DEPARTMENT: {
+          result = await collection.aggregate([
+            {
+              $project: {
+                departmentID: 1,
+                purchaseCost: 1,
+                activeCount: {
+                  $cond: [{$eq: ["$status", ACTIVE_STRING]}, 1, 0]
+                },
+                inactiveCount: {
+                  $cond: [{$ne: ["$status", ACTIVE_STRING]}, 1, 0]
+                }
+              }
+            },
+            {
+              $group: {
+                _id: "$departmentID",
+                active: {$sum: "$activeCount"},
+                inactive: {$sum: "$inactiveCount"},
+                totalCost: {$sum: "$purchaseCost"}
+              }
+            }
+          ]).toArray();
+          break;
+        }
+        case FUNCTIONALITY: {
+          result = await collection.aggregate([
+            {
+              $project: {
+                activeCount: {
+                  $cond: [{$eq: ["$status", ACTIVE_STRING]}, 1, 0]
+                },
+                inactiveCount: {
+                  $cond: [{$ne: ["$status", ACTIVE_STRING]}, 1, 0]
+                }
+              }
+            },
+            {
+              $group: {
+                _id: "null",
+                active: {$sum: "$activeCount"},
+                inactive: {$sum: "$inactiveCount"},
+                totalDevices: {$sum: 1}
+              }
+            }
+          ]).toArray();
+          break;
+        }
+        case INFLOW: {
+          //The year value should also be included in the request object in find
+          var year = find.year || null;
```

Comment:
Could variable like these be pass-in through a variable called reportSpecs.year?